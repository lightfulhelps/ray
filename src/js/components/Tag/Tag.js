// @flow
import * as React from 'react';
import classNames from 'classnames';
import { Badge, Icon } from '../../';

type Props = {
  children: string,
  className?: string,
  onRemove?: () => void,
};

const Tag = ({ children, className, onRemove, ...other }: Props) => {
  const classes = classNames(className, 'tag');

  return (
    <Badge {...other} className={classes} isPill>
      {children}
      {onRemove && <Icon className="cursor-pointer" name="close" onClick={onRemove} />}
    </Badge>
  );
};

export default Tag;
