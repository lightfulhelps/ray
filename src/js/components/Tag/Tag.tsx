import * as React from 'react';
import classNames from 'classnames';
import { Badge, Icon } from '../..';

type Props = {
  children: string;
  className?: string;
  onRemove?: () => void;
  isOutline?: boolean;
  [key: string]: any;
};

const Tag: React.FC<Props> = ({ children, className, onRemove, isOutline, ...other }: Props) => {
  const classes = classNames(className, 'tag text-sm px-2 py-1');

  return (
    <Badge
      {...other}
      theme={isOutline ? 'white' : 'dark'}
      className={classes}
      isPill
      withBorder={isOutline}
    >
      <div className="d-flex align-items-center">
        <Icon className="tag__tag-icon flex-shrink-0" name="tag" />
        {children}
        {onRemove && (
          <Icon
            className="tag__remove-icon cursor-pointer"
            data-test-id="tag-remove-icon"
            name="close"
            onClick={onRemove}
          />
        )}
      </div>
    </Badge>
  );
};

export default Tag;
