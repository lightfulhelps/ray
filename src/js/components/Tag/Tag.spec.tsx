import React from 'react';
import { shallow } from 'enzyme';
import merge from 'lodash/merge';
import Tag from './Tag';

const setup = (overrides = {}) => {
  const props = merge(
    {
      children: 'Social For Good',
    },
    overrides
  );
  const wrapper = shallow(<Tag {...props} />);

  return { wrapper, props };
};

describe('<Tag />', () => {
  it('should render', () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it('should render with a remove Icon', () => {
    const { wrapper } = setup({ onRemove: jest.fn() });

    expect(wrapper).toMatchSnapshot();
  });

  it('should handle children', () => {
    const { wrapper } = setup({ children: 'Test' });

    expect(wrapper.find('div').prop('children')).toContain('Test');
  });

  it('should handle className', () => {
    const { wrapper } = setup();

    wrapper.setProps({ className: 'custom' });

    expect(wrapper.hasClass('tag')).toBe(true);
    expect(wrapper.hasClass('custom')).toBe(true);
  });

  it('should handle onRemove', () => {
    const onRemove = jest.fn();
    const { wrapper } = setup({ onRemove });

    wrapper.find('[data-test-id="tag-remove-icon"]').simulate('click');

    expect(onRemove).toHaveBeenCalledTimes(1);
  });

  it('should pass through other props', () => {
    const { wrapper } = setup({ tabIndex: 1, id: 'test' });

    expect(wrapper.prop('tabIndex')).toEqual(1);
    expect(wrapper.prop('id')).toEqual('test');
  });
});
