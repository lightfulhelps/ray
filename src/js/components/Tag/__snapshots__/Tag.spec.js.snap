// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tag /> should render 1`] = `
<Badge
  className="tag"
  isPill={true}
>
  Social For Good
</Badge>
`;

exports[`<Tag /> should render with a remove Icon 1`] = `
<Badge
  className="tag"
  isPill={true}
>
  Social For Good
  <Icon
    className="cursor-pointer"
    name="close"
    onClick={[MockFunction]}
  />
</Badge>
`;
